### PR TITLE
Fix service_error test blinking (without cassettes)

### PR DESCRIPTION
**Pull request by coolo**
One more testsuite problem hidden by cassette

### CODE REVIEW

File: src/api/app/controllers/webui/package_controller.rb [hunk 1]
```diff
@@ -386,9 +386,6 @@ def set_file_details
       @spec_count += 1 if file[:ext] == 'spec'
     end
 
-    # check source service state
-    @package.serviceinfo.value(:error) if @package.serviceinfo
```
Comment (coolo) [inline src/api/app/controllers/webui/package_controller.rb]: did I oversee any side effect on this one?

Comment (hennevogel) [reply src/api/app/controllers/webui/package_controller.rb]: I don't think so. This 'just' fetched the file.

File: src/api/spec/models/package_spec.rb [hunk 2]
```diff
@@ -184,12 +185,18 @@
   end
 
   describe '#service_error' do
-    context 'without error' do
-      it { expect(package_with_service.service_error).to be_nil }
+    def finish_service(package)
+      sleep 0.1 while package.serviceinfo.to_hash['code'] == 'running'
```
Comment (hennevogel) [inline src/api/spec/models/package_spec.rb]: Why the sleep?

Comment (coolo) [reply src/api/spec/models/package_spec.rb]: I don't know what else do - there is no way to actively wait for the service to finish if it's broken

File: src/api/app/controllers/webui/package_controller.rb (hunk 1 above)
Comment (coolo) [reply src/api/app/controllers/webui/package_controller.rb]: Returning true since 2012 :)
16be2f5dcc21c608bf5221baba3259357690c940

Comment (hennevogel) [reply src/api/app/controllers/webui/package_controller.rb]: Returning true to void since 2012

File: src/api/spec/models/package_spec.rb (hunk 2 above)
Comment (hennevogel) [reply src/api/spec/models/package_spec.rb]: But the service is already broken in the `package_with_broken_service` factory right?

Comment (coolo) [reply src/api/spec/models/package_spec.rb]: yes, but the src server doesn't do the validation

Comment (coolo) [reply src/api/spec/models/package_spec.rb]: if we wanted to check the model function only (you would wonder why as it's nowhere called now) you could mock the backend reply directly.

**Comment by coolo** [comment]
@hennevogel I switched to stubs now

File: src/api/app/models/package.rb [hunk 3]
```diff
@@ -960,6 +960,7 @@ def jobhistory_list(project, repository, arch, options = {})
 
   def service_error(revision = nil)
     revision ||= serviceinfo.try { to_hash['xsrcmd5'] }
+    return nil unless revision
```
Comment (bgeuken) [inline src/api/app/models/package.rb]: Nitpick: You could just call return here.

**Pull request merged**